src/sentry/integrations/perforce/client.py

```diff
@@ -288,29 +351,87 @@ def get_user(self, username: str) -> dict[str, Any] | None:
                 # Check if user actually exists by verifying Update field is set
                 if not user_info.get("Update"):
                     return None
-                return {
-                    "email": user_info.get("Email", ""),
-                    "full_name": user_info.get("FullName", ""),
-                    "username": user_info.get("User", username),
-                }
+                return P4UserInfo(
+                    email=str(user_info.get("Email", "")),
+                    full_name=str(user_info.get("FullName", "")),
+                    username=str(user_info.get("User", username)),
+                )
             # User not found - return None (not an error condition)
             return None
 
     def get_changes(
-        self, depot_path: str, max_changes: int = 20, start_cl: str | None = None
-    ) -> list[dict[str, Any]]:
+        self,
+        depot_path: str,
+        max_changes: int = 20,
+        start_cl: int | None = None,
+        end_cl: int | None = None,
+    ) -> Sequence[P4ChangeInfo]:
         """
         Get changelists for a depot path.
 
+        Uses p4 changes command to list changelists.
+        API docs: https://www.perforce.com/manuals/cmdref/Content/CmdRef/p4_changes.html
+
         Args:
             depot_path: Depot path (e.g., //depot/main/...)
-            max_changes: Maximum number of changes to return
-            start_cl: Starting changelist number
+            max_changes: Maximum number of changes to return when start_cl/end_cl not specified
+            start_cl: Starting changelist number (exclusive) - returns changes > start_cl. Must be int.
+            end_cl: Ending changelist number (inclusive) - returns changes <= end_cl. Must be int.
 
         Returns:
-            List of changelist dictionaries
+            Sequence of changelist dictionaries in range (start_cl, end_cl]
+
+        Raises:
+            TypeError: If start_cl or end_cl are not integers
         """
-        return []
+        with self._connect() as p4:
+            # Validate types - changelists must be integers
+            if start_cl is not None and not isinstance(start_cl, int):
```

Comment:
I'm not entirely sure you need this since you placed the module in the stronger typing list.